core/capabilities/webapi/trigger.go

```diff
@@ -1,18 +1,322 @@
+//       signature: "...",
+//       body: {
+//         don_id: "workflow_123",
+//         payload: {
+//           trigger_id: "web-trigger@1.0.0",
+//           trigger_event_id: "action_1234567890",
+//           timestamp: 1234567890,
+//           topics: ["daily_price_update"],
+//           params: {
+//             bid: "101",
+//             ask: "102"
+//           }
+//         }
+//       }
+//     }
+//   }
+
+// from Web API Trigger Doc
+// trigger_id          - ID of the trigger corresponding to the capability ID
+// trigger_event_id    - uniquely identifies generated event (scoped to trigger_id and sender)
+// timestamp           - timestamp of the event (unix time), needs to be within certain freshness to be processed
+// topics            - [OPTIONAL] list of topics (strings) to be started by this event (affects all topics if empty)
+// workflow_owners   - [OPTIONAL] list of workflow owners allowed to receive this event (affects all workflows if empty)
+// params            - key-value pairs that will be used as trigger output in the workflow Engine (translated to values.Map)
+
+func (h *triggerConnectorHandler) HandleGatewayMessage(ctx context.Context, gatewayID string, msg *api.Message) {
+	body := &msg.Body
+	sender := ethCommon.HexToAddress(body.Sender)
+	h.lggr.Debugw("handling gateway request", "id", gatewayID, "method", body.Method, "sender", sender, "payload", body.Payload)
+	var payload workflow.TriggerRequestPayload
+	err := json.Unmarshal(body.Payload, &payload)
+	if err != nil {
+		h.lggr.Errorw("error decoding payload", "err", err)
+		return
+	}
+
+	switch body.Method {
+	case workflow.MethodWebAPITrigger:
+		h.lggr.Debugw("added MethodWebAPITrigger message", "payload", string(body.Payload))
+		// Sample payload
+		// payload: {
+		// 	trigger_id: "web-trigger@1.0.0",
+		// 	trigger_event_id: "action_1234567890",
+		// 	timestamp: 1234567890,
+		// 	topics: ["daily_price_update"],
+		// 	params: {
+		// 		bid: "101",
+		// 		ask: "102"
+		// 	}
+		// }
+
+		// I don't see anywhere there is a mapping to a different structure such as
+		// payload: {
+		//   topic: "daily_price_update"
+		// 	 params: {
+		// 		 bid: "101",
+		// 		 ask: "102"
+		// 	 }
+
+		// So this just passes it on with the expectation that that is acceptable format for the executor
+
+		wrappedPayload, _ := values.WrapMap(payload)
+
+		for _, trigger := range h.registeredWorkflows {
+
+			// TODO: Question asked in Web API trigger about checking for completed Triggers to return COMPLETED
+			// "TriggerEventID used internally by the Engine is a pair (sender, trigger_event_id).
+			// This is to protect against a case where two different authorized senders use the same event ID in their messages.
+
+			// TODO: how do we know PENDING state, that is node received the event but processing hasn't finished.
+
+			topics := payload.Topics
+
+			for _, topic := range topics {
+				if !trigger.allowedTopicsMap[topic] {
+					return
+				}
```

Comment:
for a successful trigger, do all topics in the HTTP request have to be present in `allowedTopics`? just one topic in the request being present in the workflow's allowedTopics should be sufficient right?

Reply:
That's what I'm thinking based on the description `list of topics (strings) to be started by this event`